src/Aspire.Hosting.AWS/CDK/Resources/S3ResourceExtensions.cs

```diff
@@ -0,0 +1,93 @@
+        return builder;
+    }
+
+    /// <summary>Subscribes a destination to receive notifications when an object is created in the bucket.</summary>
+    /// <param name="builder">The builder for the bucket resource.</param>
+    /// <param name="destination">The notification destination topic.</param>
+    /// <param name="filters">Filters.</param>
+    public static IResourceBuilder<IConstructResource<Bucket>> AddObjectCreatedNotification(this IResourceBuilder<IConstructResource<Bucket>> builder, IResourceBuilder<IConstructResource<ITopic>> destination, params INotificationKeyFilter[] filters )
+    {
+        builder.Resource.Construct.AddObjectCreatedNotification(new SnsDestination(destination.Resource.Construct), filters);
+        return builder;
+    }
+
+    /// <summary>Subscribes a destination to receive notifications when an object is removed from the bucket.</summary>
+    /// <param name="builder">The builder for the bucket resource.</param>
+    /// <param name="destination">The notification destination queue.</param>
+    /// <param name="filters">Filters.</param>
+    public static IResourceBuilder<IConstructResource<Bucket>> AddObjectRemovedNotification(this IResourceBuilder<IConstructResource<Bucket>> builder, IResourceBuilder<IConstructResource<IQueue>> destination, params INotificationKeyFilter[] filters )
+    {
+        builder.Resource.Construct.AddObjectCreatedNotification(new SqsDestination(destination.Resource.Construct), filters);
+        return builder;
+    }
+
+    /// <summary>Subscribes a destination to receive notifications when an object is removed from the bucket.</summary>
+    /// <param name="builder">The builder for the bucket resource.</param>
+    /// <param name="destination">The notification destination topic.</param>
+    /// <param name="filters">Filters.</param>
+    public static IResourceBuilder<IConstructResource<Bucket>> AddObjectRemovedNotification(this IResourceBuilder<IConstructResource<Bucket>> builder, IResourceBuilder<IConstructResource<ITopic>> destination, params INotificationKeyFilter[] filters )
+    {
+        builder.Resource.Construct.AddObjectRemovedNotification(new SnsDestination(destination.Resource.Construct), filters);
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds a reference of an Amazon S3 bucket to a project. The output parameters of the bucket are added to the project IConfiguration.
+    /// </summary>
+    /// <param name="builder">The builder for the resource.</param>
+    /// <param name="bucket">The Amazon S3 bucket resource.</param>
+    /// <param name="configSection">The optional config section in IConfiguration to add the output parameters.</param>
+    public static IResourceBuilder<TDestination> WithReference<TDestination>(this IResourceBuilder<TDestination> builder, IResourceBuilder<IConstructResource<Bucket>> bucket, string configSection = Constants.DefaultConfigSection)
```

Comment:
These resource level `WithReference` methods are setting up a pattern that is limiting to only one resource of the type. What happens if I need to add multiple buckets or queues. The environment variable will be computed the same for each resource type and cause collisions.

Reply:
By default it will add the resource name as part of the prefix